File: 287-find-the-duplicate-number/find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums)
    {
        unordered_map <int, int> hash;
        int size = nums.size();

        for(int i= 0;i< size;i++)
            hash[i] = 0;
        for(int i = 0;i<size;i++)
        {
            hash[nums[i]]++;
            if (hash[nums[i]] > 1)
                return (nums[i]);
        }
        return (-1);
    }
};

File: 287-find-the-duplicate-number/find-the-duplicate-number.py
class Solution(object):
    def findDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        mymap = {}
        for i in range(len(nums)):
            if nums[i] in mymap :
                return nums[i]
            mymap[nums[i]] = 1
        return -1
        